src/lib/token.test.ts

```diff
@@ -0,0 +1,313 @@
+      editState: Permissions.proofOrSignature(),
+    });
+  }
+
+  @method init() {
+    this.tokenSymbol.set(tokenSymbol);
+    this.totalAmountInCirculation.set(UInt64.zero);
+    this.maxAmountInCirculation.set(UInt64.from(100_000_000));
+  }
+
+  @method mint(receiverAddress: PublicKey, amount: UInt64) {
+    let totalAmountInCirculation = this.totalAmountInCirculation.get();
+    this.totalAmountInCirculation.assertEquals(totalAmountInCirculation);
+
+    let maxAmountInCirculation = this.maxAmountInCirculation.get();
+    this.maxAmountInCirculation.assertEquals(maxAmountInCirculation);
+
+    let newTotalAmountInCirculation = totalAmountInCirculation.add(amount);
+
+    newTotalAmountInCirculation.value
+      .lte(maxAmountInCirculation.value)
+      .assertTrue();
+
+    this.token().mint({
+      address: receiverAddress,
+      amount,
+    });
+
+    this.totalAmountInCirculation.set(newTotalAmountInCirculation);
+  }
+
+  @method burn(receiverAddress: PublicKey, amount: UInt64) {
+    this.token().burn({
+      address: receiverAddress,
+      amount,
+    });
+  }
+
+  @method send(
+    senderAddress: PublicKey,
+    receiverAddress: PublicKey,
+    amount: UInt64
+  ) {
+    this.token().send({
+      to: receiverAddress,
+      from: senderAddress,
+      amount,
+    });
+  }
+
+  @method setInvalidTokenSymbol() {
+    this.tokenSymbol.set(
+      'this-token-symbol-is-too-long-and-will-cause-an-error'
+    );
+  }
+}
+
+let zkappKey: PrivateKey;
+let zkappAddress: PublicKey;
+let zkapp: TokenContract;
+let feePayer: PrivateKey;
+
+let tokenAccount1Key: PrivateKey;
+let tokenAccount1: PublicKey;
+
+let tokenAccount2Key: PrivateKey;
+let tokenAccount2: PublicKey;
+
+// Call `setupLocal` before running each test to reset the ledger state.
+async function setupLocal() {
+  // Set up local blockchain, create zkapp keys, token account keys, deploy the contract
+  let Local = Mina.LocalBlockchain();
+  Mina.setActiveInstance(Local);
+  feePayer = Local.testAccounts[0].privateKey;
+
+  zkappKey = PrivateKey.random();
+  zkappAddress = zkappKey.toPublicKey();
+  zkapp = new TokenContract(zkappAddress);
+
+  tokenAccount1Key = Local.testAccounts[1].privateKey;
+  tokenAccount1 = tokenAccount1Key.toPublicKey();
+
+  tokenAccount2Key = Local.testAccounts[2].privateKey;
+  tokenAccount2 = tokenAccount2Key.toPublicKey();
+
+  (
+    await Mina.transaction(feePayer, () => {
+      Party.fundNewAccount(feePayer);
+      zkapp.deploy({ zkappKey });
+      zkapp.init();
+    })
+  ).send();
+}
+
+describe('Token', () => {
+  beforeAll(async () => await isReady);
+
+  afterAll(() => setTimeout(shutdown, 0));
+
+  describe('Create existing token', () => {
+    beforeEach(async () => {
+      await setupLocal();
+    });
+
+    it('should have a valid custom token id', async () => {
+      const tokenId = zkapp.token().id;
+      const expectedTokenId = new Token({ tokenOwner: zkappAddress }).id;
+      expect(tokenId).toBeDefined();
```

Comment:
could delete this line `expect(tokenId).toBeDefined();` given the following assertion, no? Or is there a chance it returns undefined?